### PR TITLE
Dev yanis

### PR DESCRIPTION
**Pull request by Yanis-F**
All necessary changes to make if work with emscripten.
This also includes the changes from my last PR from which I canceled CI workflow

### CODE REVIEW

File: hdr/libccc/format.h [hunk 1]
```diff
@@ -43,6 +43,7 @@
 #endif
 //!@}
 
+#include <inttypes.h>
```
Comment (LexouDuck) [inline hdr/libccc/format.h:46]: I will probably need to add some code to define the `PRIxx` in the case when the `__NOSTD__` macro is active. (Notice how any standard header included will always be in a `#ifndef __NOSTD__` block)

**Comment by LexouDuck** [comment]
Good compatibility changes - I will merge this now, and add the `__NOSTD__` fixes (mentioned in a comment above) soon.

File: hdr/libccc/format.h (hunk 1 above)
Comment (Yanis-F) [reply hdr/libccc/format.h:46]: right, that is a good point

Comment (LexouDuck) [reply hdr/libccc/format.h:46]: ok, done via 9db813490cf69cc29db86711776b5aa32c25c20c

**Pull request merged**